### PR TITLE
test: add test for pkg/app

### PR DESCRIPTION
**Pull request by chaoranz758**
#### What type of PR is this?

 test

#### Check the PR title.

- [x] This PR title match the format: `<type>(optional scope): <description>`.
- [x] The description of this PR title is user-oriented and clear enough for others to understand.
- [x] Attach the PR updating the user documentation if the current PR requires user awareness at the usage level. [User docs repo](https://github.com/cloudwego/cloudwego.github.io).

#### (Optional) Translate the PR title into Chinese.

提升pkg/app的单元测试覆盖率

#### (Optional) More detail description for this PR(en: English/zh: Chinese).

en:Supplement unit test for pkg/app from 70.87% to 84.8%
zh(optional):提升pkg/app的单元测试覆盖率70.87%到84.8%

#### (Optional) Which issue(s) this PR fixes:
part of #257

### CODE REVIEW

**Comment by welkeyever** [comment]
Better add some info - including the coverage  changes - in the description of PR.

**Comment by chaoranz758** [comment]
> Better add some info - including the coverage changes - in the description of PR.

done

File: pkg/app/client/client_test.go [hunk 1]
```diff
@@ -2267,7 +2267,7 @@ func TestClientDoWithDialFunc(t *testing.T) {
 
 func TestClientState(t *testing.T) {
 	opt := config.NewOptions([]config.Option{})
-	opt.Addr = ":11000"
+	opt.Addr = ":10036"
```
Comment (li-jin-gou) [inline pkg/app/client/client_test.go:2270]: 为什么要修改 addr ～

Comment (chaoranz758) [reply pkg/app/client/client_test.go]: 11000端口是华硕电脑Aura Wallpaper Service.exe的运行端口（我的电脑是这样），TestLoadHTMLGlob中修改的端口类似，在我的电脑中是一个驱动程序的运行端口。但我最开始修改的端口又与其他的test冲突，因此我又重新修改了端口并提交

File: pkg/app/context_test.go [hunk 2]
```diff
@@ -436,6 +455,21 @@ tailfoobar`
...
+		t.Fatalf("unexpected error: %s", err)
+	}
+	fileInfo, err := os.Stat("TODO")
+	if err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	}
+	assert.DeepEqual(t, "TODO", fileInfo.Name())
+	assert.DeepEqual(t, f.File["fileaaa"][0].Size, fileInfo.Size())
+	err = os.Remove("TODO")
+	if err != nil {
```
Comment (welkeyever) [inline pkg/app/context_test.go:469]: lines 469-470 could be replaced like:
```
assert.Nil(t, err)
```

File: pkg/app/context_test.go [hunk 3]
```diff
@@ -436,6 +455,19 @@ tailfoobar`
 		}
 	}
 
+	err = ctx.SaveUploadedFile(f.File["fileaaa"][0], "TODO")
+	if err != nil {
```
Comment (welkeyever) [inline pkg/app/context_test.go:459]: AA, you may check and fix them all.

Comment (chaoranz758) [reply pkg/app/context_test.go:459]: ok

**Review by welkeyever — DISMISSED** [review]
LGTM, thx~

**Review by li-jin-gou — APPROVED** [review]

**Review by welkeyever — APPROVED** [review]

**Pull request merged**